packages/flutter/lib/src/widgets/text.dart

```diff
@@ -1401,3 +1521,21 @@ class _SelectableTextContainerDelegate extends MultiSelectableSelectionContainer
     super.didChangeSelectables();
   }
 }
+
+class _TextSpanContentRange extends SelectedContentRange<TextSpan> {
+  _TextSpanContentRange({
+    this.start = -1,
+    this.end = -1,
+    super.selectableId,
+    required super.content,
+    super.children,
+  });
+
+  @override
+  int get startOffset => start;
+  final int start;
+
+  @override
+  int get endOffset => end;
+  final int end;
+}
```

Comment:
I wonder if you could effectively get the functionality of _TextSpanContentRange with a type parameter instead of a subclass? Since the only interesting thing here is really the TextSpan type.

Reply:
I have made `SelectedContentRange` non-abstract so we can get rid of these sublasses.